src/meep.hpp

```diff
@@ -1346,11 +1357,12 @@ class fields_chunk {
   structure_chunk *new_s;
   structure_chunk *s;
   const char *outdir;
+  int chunk_idx;
```

Comment:
Why was it necessary to add a new member variable `chunk_idx` to the `fields_chunk` class? It doesn't seem to be actually used for anything. There is a member function `n_proc()` which already returns the chunk ID (equivalent to the MPI rank) and thus adding a `chunk_idx` variable seems redundant.

Reply:
Can't a proc have multiple chunks?

Reply:
Yes for the PML regions but still `chunk_idx` doesn't seem to be used for anything.

Reply:
``chunk_idx`` is first used in ``dft_near2far::near_sourcedata``, as the ``fc_idx`` in the structure ``sourcedata``. That information is later used in`` fields::add_srcdata`` to find the right field chunk to add source.

Reply:
`dft_near2far::near_sourcedata` seems to be creating a `sourcedata` object for every `dft_chunk` and since `dft_chunk`s are only ever created in *non-PML* regions it should be unnecessary to add a separate `chunk_idx` member variable to the `fields_chunk` class since `n_proc()` already provides this information.